packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit/utils.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  generateUUID,
+  formatDate,
+  arrayToRecord,
+} from '@openmsupply-client/common';
+import { InboundShipmentItem, InboundShipmentRow } from '../../../../types';
+import { recalculateSummary } from '../../../../OutboundShipment/DetailView/reducer';
+import { Item } from '@openmsupply-client/mock-server/src/data';
+
+export const createInboundShipmentBatch = (
+  inboundItem: InboundShipmentItem,
+  seed?: InboundShipmentRow
+): InboundShipmentRow => {
+  const id = generateUUID();
+  const row = {
+    id,
+    numberOfPacks: 0,
+    stockLineId: '',
+    invoiceId: '',
+    itemId: '',
+    note: '',
+    costPricePerPack: 0,
+    expiryDate: formatDate(new Date()),
+    itemCode: '',
+    itemName: '',
+    packSize: 1,
+    sellPricePerPack: 0,
+    ...seed,
+    update: (key: string, value: string) => {
```

Comment:
actually its just because the function is optional. When it isn't optional it does error if you dont have the function. Maybe need to make sure it's required!

Reply:
I thought it was a good approach for cell updating - as you say, allows the same columns to be used in different contexts and the calling component can customise as needed